documentation/bolt_types_reference.md

```diff
@@ -23,6 +23,29 @@ The following functions are available to `ApplyResult` objects.
 | `target` | `Target` | The target the result is from. |
 | `to_data` | `Hash` | A serialized representation of `ApplyResult`. |
 
+## `ResourceInstance`
+
+`ResourceInstance` objects are used to store the observed and desired state of a target's resource and to track events for the resource. These objects do not modify or interact with a target's resources.
```

Comment:
I think it'd be useful to document an example of using this in a plan. I think just in the resource type page would actually be good since this is experimental, but writing_plans.md would also be appropriate

Reply:
👍 I'll write up some pre-docs for this once the API is a little more solidified.